R/shape_measures.R

```diff
@@ -25,46 +25,88 @@
-  if (identical(type, "LTcv"))   S = c(LTcv = LifetableCV(lt$nx, lt$nax, lt$dx, lt$ex))
+  if (identical(type, "LTentr"))
+    S = c(LTentr = LifetableEntropy(lt$nax, lt$nx, lt$ndx, lt$lt, lt$ex))
+  if (identical(type, "LTgini"))
+    S = c(LTgini = LifetableGini(lt$nax, lt$nx, lt$lx, lt$ex))
+  if (identical(type, "LTcv"))
+    S = c(LTcv = LifetableCV(lt$nax, lt$nx, lt$dx, lt$lx, lt$ex))
   if (identical(type, "all")) {
-    S = c(LTentr = LifetableEntropy(lt$nax, lt$nx, lt$ndx, lt$ex),
-          LTgini = LifetableGini(lt$nx, lt$nax, lt$lx, lt$ex),
-          LTcv   = LifetableCV(lt$nx, lt$nax, lt$dx, lt$ex))
+    S = c(LTentr = LifetableEntropy(lt$nax, lt$nx, lt$ndx, lt$lx, lt$ex),
+          LTgini = LifetableGini(lt$nax, lt$nx, lt$lx, lt$ex),
+          LTcv   = LifetableCV(lt$nax, lt$nx, lt$dx, lt$lx, lt$ex))
   }
   return(S)
 }
 
 # Lifetable Entropy -------------------------------------------------------
 
-# Average Years of Life Lost due to Death in Age x
-eDaggerx <- function (nax, nx, ex) {
+#' Average Years of Life Lost due to Death in Age x
+#'
+#' @keywords internal
+eDaggerx <- function (nax, nx, nlx, ex) {
   nAx = nax/nx
-  edx = nAx * c(ex[-1], 0) + (1 - nAx) * ex
+  edx = (nAx * c(ex[-1], 0) + (1 - nAx) * ex) / nlx[1]
+  edx[length(edx)] <- max(edx[length(edx)], 0)
   return(edx)
 }
 
-# Total Life Years Lost due to Death
-EDagger <- function (nax, nx, ndx, ex) {
-  edx = eDaggerx(nax, nx, ex)
+#' Total Life Years Lost due to Death
+#'
+#' @keywords internal
+EDagger <- function (nax, nx, ndx, nlx, ex) {
+  edx = eDaggerx(nax, nx, nlx, ex)
   ed = sum(ndx*edx)
   return(ed)
 }
 
-# Life Table Entropy
-LifetableEntropy <- function (nax, nx, ndx, ex) {
-  ed = EDagger(nax, nx, ndx, ex)
+#' Life Table Entropy
+#'
+#' @keywords internal
+LifetableEntropy <- function (nax, nx, ndx, nlx, ex) {
+  ed = EDagger(nax, nx, ndx, nlx, ex)
   H = 1 - ed / ex[1]
   return(H)
 }
 
-# Life Table Gini-Coefficient
-LifetableGini <- function (nx, nax, lx, ex) {
-  return(NA)
+#' Life Table Gini-Coefficient
+#'
+#' @seealso LifetableGini
+#' @keywords internal
+LifetableGini <- function (nax, nx, nlx, ex) {
+  nAx <- nax/nx
+  nlx <- nlx/nlx[1]
+  Gx = nAx * c(nlx[-1], 0) + (1 - nAx) * nlx
+  Gx = Gx^2 * nx
+  Gx[length(Gx)] <- max(Gx[length(Gx)], 0)
+  G = 2/ex[1]*sum(Gx) - 1
+  return(G)
+}
+
+#' Life Table Gini-Coefficient
+#'
+#' Another discrete formulation of the Gini-Coeffcient
+#' @seealso LifetableGini
+#' @keywords internal
+LifetableGini2 <- function (nax, nx, nlx, ex) {
+  nAx = nax/nx
+  nlx = nlx/nlx[1]
+  nlx_1 = c(nlx[-1], 0)
+  Gx = nlx_1^2 + nAx*(nlx^2 - nlx_1^2)
+  Gx = Gx * nx
+  G = 1 - 1/ex[1] * sum(Gx)
+  return(1-G)
 }
 
-# Life Table Coefficient of Variation
-LifetableCV <- function (nx, nax, dx, ex) {
-  return(NA)
+#' Life Table Coefficient of Variation
+#'
+#' @keywords internal
+LifetableCV <- function (nax, nx, ndx, nlx, ex) {
```

Comment:
You are quite right about the `ndx` notation instead of `dx`. I was wrong here. dx would be the density of deaths whereas `ndx` is the actual number of deaths in age interval `[x, x+n)` of the life table population.